MyOnboarding/src/MyOnboardingApp.Api/App_Start/DependenciesConfig.cs

```diff
@@ -1,40 +1,96 @@
+using System.Linq;
 using System.Web.Http;
+using MyOnboardingApp.Api.Configuration;
 using MyOnboardingApp.Api.DependencyResolvers;
 using MyOnboardingApp.Api.UrlLocation;
 using MyOnboardingApp.ApiServices;
 using MyOnboardingApp.Contracts.Registration;
+using MyOnboardingApp.Contracts.Repository;
 using MyOnboardingApp.Contracts.Urls;
 using MyOnboardingApp.Database;
+using MyOnboardingApp.Services;
+using MyOnboardingApp.Validation;
 using Unity;
 using Unity.Lifetime;
 
-[assembly: InternalsVisibleTo("MyOnboardingApp.Api.Tests")]
 
 namespace MyOnboardingApp.Api
 {
     internal static class DependenciesConfig
     {
+        private static readonly IList<IBootstrapper> s_bootstrappers = new List<IBootstrapper>();
+
+
         public static void Register(HttpConfiguration config)
-        {
-            var container = new UnityContainer();
-            RegisterAllDependencies(container);
-            config.DependencyResolver = new DependencyResolver(container);
-        }
+            => new UnityContainer()
+                .RegisterAllDependencies()
+                .ValidateConfiguration()
+                .ConfigureDependencyResolver(config);
 
 
-        internal static void RegisterAllDependencies(IUnityContainer container) 
+        internal static IUnityContainer RegisterAllDependencies(this IUnityContainer container)
             => container
                 .RegisterDependency<ApiServicesBootstrapper>()
                 .RegisterDependency<DatabaseBootstrapper>()
-                .RegisterType<IRoutesConfig, RoutesConfig>(new HierarchicalLifetimeManager());
+                .RegisterDependency<ServicesBootstrapper>()
+                .RegisterDependency<ValidationBootstrapper>()
+                .RegisterType<IRoutesConfig, RoutesConfig>(new HierarchicalLifetimeManager())
+                .RegisterType<IDatabaseConnection, DatabaseConnection>(new HierarchicalLifetimeManager());
+
+
+        private static IUnityContainer RegisterDependency<TDependency>(this IUnityContainer container)
+            where TDependency : IBootstrapper, new()
+            => new TDependency()
+                .StoreInstance()
+                .Register(container);
+
+
+        private static TDependency StoreInstance<TDependency>(this TDependency bootstrapper)
+            where TDependency : IBootstrapper, new()
+        {
+            s_bootstrappers.Add(bootstrapper);
+
+            return bootstrapper;
+        }
 
 
-        private static IUnityContainer RegisterDependency<TDependency>(this IUnityContainer container) where TDependency : IBootstrapper, new()
+        private static IUnityContainer ValidateConfiguration(this IUnityContainer container)
         {
-            var dependency = new TDependency();
-            dependency.Register(container);
+            if (!s_bootstrappers.Any())
+                throw new Exception("Bootstrappers have to be registered before checking validity of their configuration.");
```

Comment:
Maybe just a slight change in the message to concentrate it more on a possibility the bootstrappers get lost rather than order of extension methods changed (what I deem less likely):
```suggestion
                throw new Exception("No bootstrappers have been registered before checking the validity of their configuration.");
```